### PR TITLE
Use gtest framework for Configuration subsystem

### PR DESCRIPTION
**Pull request by guanyung-rs**
> ### Motivate of the pull request
> - [ ] To address an existing issue. If so, please provide a link to the issue: <issue id>
> - [ ] Breaking new feature. If so, please describe details in the description part.

Follow up work from this PR https://github.com/os-fpga/FOEDAG/pull/1049
Use google test framework for Configuration subsystem tests

> ### Describe the technical details
> #### What is currently done? (Provide issue link if applicable)
> 
> 
> 
> 
> 
>
> #### What does this pull request change?
> 
> 
> 
> 
> 

> ### Which part of the code base require a change
> 
> - [ ] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.

### CODE REVIEW

File: src/Configuration/CFGCommon/Test/CFGCommon_test.cpp [hunk 1]
```diff
@@ -21,18 +21,25 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "CFGCommon.h"
 
+#include <gtest/gtest.h>
```
Comment (KochynVolodymyr) [inline src/Configuration/CFGCommon/Test/CFGCommon_test.cpp:24]: We have another folder for the unit tests: `tests/unittest/`

File: src/Configuration/CFGCompiler/Test/CFGCompiler_test.cpp [hunk 2]
```diff
@@ -1,9 +1,9 @@
 /*
-Copyright 2021 The Foedag team
+Copyright 2023 The Foedag team
```
Comment (KochynVolodymyr) [inline src/Configuration/CFGCompiler/Test/CFGCompiler_test.cpp:2]: We have another folder for the unit tests: `tests/unittest/`

**Review by ravic-rs — CHANGES_REQUESTED** [review]
@guanyung-rs as mentioned in previous PR, please move the gtests to 
https://github.com/os-fpga/FOEDAG/tree/main/tests/unittest

**Comment by guanyung-rs** [comment]
Ok. I will move them to the tests/unittest folder

**Pull request closed without merging**